Feat: Newspack Avatar block

AKSHAT2802 opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Closes https://app.asana.com/1/26890605006346/project/1209276474023429/task/1209276476021332?focus=true

### How to test the changes in this Pull Request:
- Add the Newspack Avatar block to a post or page in the WordPress editor.
- Test the block with single and multiple authors (if Co-Authors Plus is active).
- Adjust the avatar size and toggle the "Link to user profile" option in the block settings.
- Verify that the duotone filters and border styles are applied correctly.
- Check the front end to ensure avatars render as expected.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

AKSHAT2802 committed:
feat: add Avatar block with customizable options and styles

AKSHAT2802 committed:
fix: improve avatar image rendering and style handling

thomasguillot committed:
feat: update icon, description, and togglecontrol label

thomasguillot committed:
fix: stacked inline style

thomasguillot commented:
Thanks for this @AKSHAT2802.

A few things:

* I'd suggest removing the resize box. When there are multiple avatars it looks a bit odd and the behaviour is a bit clumsy.
* I don't see the SCSS on the front-end 
* When enabling "Link to author archive", we loose the negative on the stacked style

AKSHAT2802 committed:
refactor: simplify Avatar component by removing ResizableBox and consolidating rendering logic

AKSHAT2802 commented:
Thanks for the review, @thomasguillot!
I've fixed the issue with the "Link to author archive" styles and implemented your suggestion to replace the ResizableBox with a simpler wrapper.
Let me know if there's anything else you'd like adjusted.

thomasguillot committed:
feat: add mask-image to stacked style

thomasguillot reviewed (dismissed):
I made some more small changes https://github.com/Automattic/newspack-plugin/pull/3921/commits/a726414428a2057bd01e0d4c2c834c439d260338

Design ✅ 

![](https://github.com/user-attachments/assets/74255f95-7d3c-4bac-b6cd-ab9541d1fbf0)

I'm wondering what's this `onerror` on my image here.

thomasguillot commented:
> I'm wondering what's this onerror on my image here.

Never mind. I can see this in `trunk` as well.

thomasguillot reviewed (commented):
Waiting on @Automattic/newspack-product to fully approve this PR.

thomasguillot committed:
Merge trunk into feat/newspack-avatar-block

thomasguillot committed:
feat: import colors and use for icon's foreground

leogermani commented:
Not sure if I did something wrong, but it didn't work for multiple authors.

I'm using a block called "avatar". Is this the correct one? if I search by "newspack/avatar" I don't find anything...

Should we mark this block in a way it can be recognized as a newspack block? I think all the other have blue icons (cc @thomasguillot )

AKSHAT2802 commented:
Hii @leogermani 
You're probably seeing the default core Avatar block with a black icon. The Newspack one has a blue icon.

Yes, I agree — I think it would really help to mark it more clearly as a Newspack block. Right now it’s easy to confuse it with the core Avatar block, especially since the names are the same.

![image](https://github.com/user-attachments/assets/fe949db1-6cb4-4a42-9657-1907d5a5ed97)

thomasguillot commented on src/blocks/avatar/index.js:
```diff
@@ -0,0 +1,40 @@
+		src: icon,
+		foreground: colors['primary-400'],
+	},
+	keywords: [__('author', 'newspack-plugin'), __('user', 'newspack-plugin'), __('profile', 'newspack-plugin')],
```
Maybe we should add "newspack" as a keyword so it's easier to find

AKSHAT2802 committed:
fix(avatar-block): add newspack as keyword for avatar block

leogermani commented:
Ok, I see the block now. But it's also not working with multiple authors.

Here's on the Template editor

![image](https://github.com/user-attachments/assets/a9b3268c-2303-49e0-bde3-9bae9a8f131d)

Here's on the front end

![image](https://github.com/user-attachments/assets/52d44ebe-bde6-46f0-8b62-8c61b1934294)

AKSHAT2802 commented:
As we are editing the template -- which will not have a post ID and co-authors -- it will fall back to a default avatar. 

However, once the template is edited and saved successfully, the expected avatars render correctly for all posts on the front end.
It is working well for my post edit screen, too.

Screen recording: 

https://github.com/user-attachments/assets/07a7eeb4-41a5-4baf-b17e-de2049552676

leogermani commented:
Hi @AKSHAT2802 , I understand that. But for some reason in my site it only renders one avatar. Did you see the screenshots? I did the same thing... Not sure if I'm doing something wrong or if there's something wrong/missing in my site

AKSHAT2802 commented:
Hii @leogermani 

Thanks for looking into this. I did try testing the block on a fresh WordPress install multiple times.
Here are the exact steps I followed:
- Cloned the Newspack plugin & Newspack Block theme, and activated it.
- Installed the Co-Authors Plus plugin
- Checkout to this branch.
- Create a post with multiple authors.
- Switch to template editor from the post editor / customise the template directly.
- Add avatar block and check posts on the frontend.

Could you try testing once more using the same setup? If the issue persists, we can hop on a quick huddle to debug and resolve it together.

Let me know what works best for you! 🙇
